test/e2e/backup-resource/deletion.go

```diff
@@ -0,0 +1,162 @@
+func runBackupDeletionTests(client TestClient, veleroCLI, providerName, veleroNamespace, backupName, backupLocation string,
+	useVolumeSnapshots bool, registryCredentialFile, region, bslPrefix, bslConfig string) error {
+
+	oneHourTimeout, _ := context.WithTimeout(context.Background(), time.Minute*60)
+
+	if err := CreateNamespace(oneHourTimeout, client, deletionTest); err != nil {
+		return errors.Wrapf(err, "Failed to create namespace %s to install Kibishii workload", deletionTest)
+	}
+	defer func() {
+		if err := DeleteNamespace(context.Background(), client, deletionTest, true); err != nil {
+			fmt.Println(errors.Wrapf(err, "failed to delete the namespace %q", deletionTest))
+		}
+	}()
+
+	if err := KibishiiPrepareBeforeBackup(oneHourTimeout, client, providerName, deletionTest, registryCredentialFile); err != nil {
+		return errors.Wrapf(err, "Failed to install and prepare data for kibishii %s", deletionTest)
+	}
+	err := ObjectsShouldNotBeInBucket(VeleroCfg.CloudProvider, VeleroCfg.CloudCredentialsFile, region, VeleroCfg.BSLBucket, VeleroCfg.BSLPrefix, VeleroCfg.BSLConfig, backupName, BackupObjectsPrefix, 1)
+	if err != nil {
+		return err
+	}
+	if err := VeleroBackupNamespace(oneHourTimeout, veleroCLI, veleroNamespace, backupName, deletionTest, backupLocation, useVolumeSnapshots); err != nil {
+		// TODO currently, the upgrade case covers the upgrade path from 1.6 to main and the velero v1.6 doesn't support "debug" command
+		// TODO move to "runDebug" after we bump up to 1.7 in the upgrade case
+		VeleroBackupLogs(context.Background(), VeleroCfg.UpgradeFromVeleroCLI, veleroNamespace, backupName)
+		return errors.Wrapf(err, "Failed to backup kibishii namespace %s", deletionTest)
+	}
+
+	if providerName == "vsphere" && useVolumeSnapshots {
+		// Wait for uploads started by the Velero Plug-in for vSphere to complete
+		// TODO - remove after upload progress monitoring is implemented
+		fmt.Println("Waiting for vSphere uploads to complete")
+		if err := WaitForVSphereUploadCompletion(oneHourTimeout, time.Hour, deletionTest); err != nil {
+			return errors.Wrapf(err, "Error waiting for uploads to complete")
+		}
+	}
+	err = ObjectsShouldBeInBucket(VeleroCfg.CloudProvider, VeleroCfg.CloudCredentialsFile, region, VeleroCfg.BSLBucket, bslPrefix, bslConfig, backupName, BackupObjectsPrefix)
+	if err != nil {
+		return err
+	}
+	err = DeleteBackupResource(context.Background(), veleroCLI, backupName)
+	if err != nil {
+		return err
+	}
+	err = ObjectsShouldNotBeInBucket(VeleroCfg.CloudProvider, VeleroCfg.CloudCredentialsFile, region, VeleroCfg.BSLBucket, bslPrefix, bslConfig, backupName, BackupObjectsPrefix, 5)
+	if err != nil {
+		fmt.Println(errors.Wrapf(err, "Failed to get object from bucket %q", backupName))
+		return err
+	}
+	backupName = "backup-1-" + UUIDgen.String()
+	if err := VeleroBackupNamespace(oneHourTimeout, veleroCLI, veleroNamespace, backupName, deletionTest, backupLocation, useVolumeSnapshots); err != nil {
```

Comment:
Why do we need to backup again?

Reply:
Because the 1st backup was deleted locally for verification that backup can be deleted from object store right after local deletion.